ruby/vampires.rb

```diff
@@ -23,18 +23,23 @@
 
 
 
-	if allergies =! "done"
+	until allergies == "done" or allergies == "sunshine"
 
 		puts "Do you have any suspicious allergies?"
 		allergies = gets.chomp
 
 		if allergies == "sunshine"
 			puts "Probably a vampire"
+			allergies = "done"
 		end
 
 		
 	end
 
+
+
+	p "Actually, never mind! What do these questions have to do with anything? Let's all be friends."
+
 			
 
 
```

Comment:
Consider the optimum condition when running your code.  What can we do to run only necessary processes?  We know that if the employee has Drac or Fang in their name then they are definitely a vampire.  If that is the case, do we need to run any of the other checks?

Consider:

```
if name_matches_vampire_params
  "Definitely a vampire"
else
  if #other checks
    "Something"
  elsif #more conditions...

  else
  end
end
```